pkg/worker/workflow.go

```diff
@@ -402,6 +398,13 @@ func (w *worker) TriggerPipelineWorkflow(ctx workflow.Context, param *TriggerPip
 			}
 			componentRunFutures = append(componentRunFutures, workflow.ExecuteActivity(minioCtx, w.UploadComponentOutputsActivity, futureArgs[idx]))
 		}
+
+		for idx := range futures {
+			// There is time difference between the workflow memory update and upload component inputs activity.
```

Comment:
❓ Does this mean that, if the component activity fails, we won't have access to the component input in the run logs? If so:
- I'd document that scenario here.
- This means that `ComponentActivity` depends on `UploadComponentInputsActivity` if we want the full log support, and that we shouldn't use futures for this activity. We can modify this post-release.

Reply:
> Does this mean that, if the component activity fails, we won't have access to the component input in the run logs?

No, it doesn't mean it.
It means `UploadComponentInputsActivity` should be executed after `ComponentActivity`.

Accurately, we should execute `UploadComponentInputsActivity` after the workflow memory is set.

However, because of unknown reason, probably worker execution sequence problem, it will be safer to execute `UploadComponentInputsActivity` after `UploadComponentOutputsActivity`.

I tested to set `UploadComponentInputsActivity` after `ComponentActivity`. But, somehow, the workflow memory sometimes did not set when I executed it after `ComponentActivity`.

Reply:
Okay, then this should be investigated (otherwise the order of the activities is a bit puzzling) but I'm not against merging the current code.

Reply:
> otherwise the order of the activities is a bit puzzling

Yes, I also want to know the reason! 
I might dig in Temporal document and do more experiments. 😃 

Thanks for pointing this out.